src/algebra/iterate_hom.lean

```diff
@@ -109,3 +115,19 @@ theorem iterate_map_gsmul (n : ℕ) (m : ℤ) (x : R) :
 f.to_add_monoid_hom.iterate_map_gsmul n m x
 
 end ring_hom
+
+@[simp] lemma mul_left_iterate [monoid M] (a : M) (n : ℕ) : ((*) a^[n]) = (*) (a^n) :=
```

Comment:
I had a hard time parsing this one, because priorities are not obvious (and they feel wrong to me as things are written). What about
```suggestion
@[simp] lemma mul_left_iterate [monoid M] (a : M) (n : ℕ) : ((*) a)^[n] = (*) (a^n) :=
```
(the parentheses around `(*) a` are not needed, but they help)

Same thing for the next lemma.

Reply:
Done.